sdk/python/feast/infra/online_stores/remote.py

```diff
@@ -60,7 +60,77 @@ def online_write_batch(
         ],
         progress: Optional[Callable[[int], Any]],
     ) -> None:
-        raise NotImplementedError
+        """
+        Writes a batch of feature rows to the remote online store via the remote API.
+        """
+        assert isinstance(config.online_store, RemoteOnlineStoreConfig)
+        config.online_store.__class__ = RemoteOnlineStoreConfig
+
+        # Restructure data into a columnar dictionary format for the 'df' key
+        columnar_data: Dict[str, List[Any]] = {}
+
+        # Collect all unique entity key names and feature names
+        all_keys = set()
```

Comment:
I think all_keys collection is not needed, you can simply populate in the below loop once defaultdict is initialized above.